src/languages/java.rs

```diff
@@ -68,59 +104,51 @@ impl JavaProcessor {
         None
     }
 
-    /// Extracts the container (class or interface) with the method declaration and body.
-    ///
-    /// @example
-    ///
-    /// ```java
-    /// public class MyClass {
-    ///     public void method() {
-    ///         methodBody();
-    ///     }
-    /// }
-    /// ```
-    ///
-    /// is split into:
-    ///
-    /// ```java
-    /// public class MyClass {
-    ///     public void method() {
-    /// ```
-    /// and
-    ///
-    /// ```java
-    ///         methodBody();
-    ///     }
-    /// }
-    /// ```
-    fn extract_container_with_method(
+    /// Extracts the method with its full context from nested classes/interfaces
+    fn extract_method_with_context(
         &self,
-        content: &str,
-        container_node: Node,
         method_node: Node,
+        content: &str,
+        stack: &[(Node, String)],
     ) -> (String, String) {
-        // Find the container declaration start (including JavaDoc/comments before it)
-        let container_start = self.find_container_start_with_comments(container_node);
-        let container_body_start = self.find_container_body_start(container_node);
+        let mut declaration = String::new();
 
-        if let Some(body_start) = container_body_start {
-            let mut declaration = String::new();
+        // Build the full context from the stack
+        for (i, (container_node, _container_name)) in stack.iter().enumerate() {
+            if i > 0 {
+                declaration.push('\n');
+            }
 
-            // Add everything from the container start (including JavaDoc) up to the opening brace
-            let container_decl = &content[container_start..body_start];
-            declaration.push_str(container_decl);
-            declaration.push('\n');
+            // Add container declaration with proper indentation
+            let container_start = self.find_container_start_with_comments(*container_node);
+            let container_body_start = self.find_container_body_start(*container_node);
 
-            // Split method into declaration and body using AST
-            let (method_declaration, method_body) =
-                self.split_method_declaration_and_body(method_node, content);
+            if let Some(body_start) = container_body_start {
+                let container_decl = &content[container_start..body_start + 1];
+                // Use the original indentation from the source - only adjust if this is a nested container
+                if i == 0 {
+                    // First container - use original indentation as-is
+                    declaration.push_str(container_decl);
+                } else {
+                    // Nested container - add base indentation to the whole chunk
+                    let base_indent = "    ".repeat(i);
```

Comment:
The indentation must be taken from the code file, we should not assume a 4-space indent. Explore TreeSitter's Node structure, it should contain the property that tells us 'the position of a node within the line' which is roughly what indentation is, if we assume that the node is the first on the line (and if it isn't, that's fine)